### PR TITLE
[backport] 7.x.x/backports/yamato/ruamel refactor

### PR DESCRIPTION
**Pull request by kivistikliis**
---
### Purpose of this PR
Backporting refactored Yamato jobs, which are now generated via Python/ruamel script. Original backport PR is #598, but as it has received several updates since then, the refactor was taken straight from the latest master instead of cherrypicking.

---
### Testing status

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/7.x.x%252Fbackports%252Fyamato%252Fruamel-refactor

### CODE REVIEW

**Comment by kivistikliis** [comment]
Some jobs and/or dependencies on 7.x.x are not in sync with master, the PR will be marked ready when its decided what to sync and what not

File: .yamato/_abv.yml [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+      -  path: .yamato/_abv.yml#smoke_test_Standalone_2019.3
+         rerun: on-new-revision
+      -  path: .yamato/_abv.yml#smoke_test_playmode_2019.3
+         rerun: on-new-revision
+      -  path: .yamato/_abv.yml#smoke_test_editmode_2019.3
+         rerun: on-new-revision
+      -  path: .yamato/universal-android-opengles3.yml#Universal_Android_OpenGLES3_Standalone_2019.3
+         rerun: on-new-revision
+      -  path: .yamato/universal-android-vulkan.yml#Universal_Android_Vulkan_Standalone_2019.3
+         rerun: on-new-revision
```
Comment (sophiaaar) [inline .yamato/_abv.yml:106]: android targets arent running on 7.x.x/release. i think we should exclude them from this backport PR and add them back in at another time

File: .yamato/all-hdrp.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      -  path: .yamato/hdrp-osx-metal.yml#HDRP_OSX_Metal_editmode_2019.3
+         rerun: on-new-revision
+      -  path: .yamato/hdrp-osx-metal.yml#HDRP_OSX_Metal_playmode_2019.3
+         rerun: on-new-revision
+      -  path: .yamato/hdrp-linux-vulkan.yml#HDRP_Linux_Vulkan_editmode_2019.3
+         rerun: on-new-revision
+      -  path: .yamato/hdrp-linux-vulkan.yml#HDRP_Linux_Vulkan_playmode_2019.3
+         rerun: on-new-revision
+      -  path: .yamato/hdrp_dxr-win-dx12.yml#HDRP_DXR_Win_DX12_playmode_2019.3
+         rerun: on-new-revision
```
Comment (sophiaaar) [inline .yamato/all-hdrp.yml:34]: i'm not sure if DXR will work on 2019.3 - @theo-pnv could you look into this?

Comment (theo-pnv) [reply .yamato/all-hdrp.yml:34]: Yes, on it

Comment (theo-pnv) [reply .yamato/all-hdrp.yml:34]: So I don't know if DXR works for 2019.3 but in any case, it's not part of the All HDRP CI pipeline, for 7.x.x and 8.x.x (see here https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/7.x.x%252Frelease/.yamato%252Fupm-ci-hdrp.yml%2523All_HDRP_2019.3/2502558/job/pipeline VS for master https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/master/.yamato%252Fall-hdrp.yml%2523All_HDRP_trunk/2498833/job/pipeline)
(I hope I'm looking at the right places)

File: .yamato/_abv.yml (hunk 1 above)
Comment (kivistikliis) [reply .yamato/_abv.yml]: solved this with the last commit

File: .yamato/all-hdrp.yml (hunk 2 above)
Comment (kivistikliis) [reply .yamato/all-hdrp.yml]: removed it with the last commit!

**Comment by kivistikliis** [comment]
moved to #1031

**Pull request closed without merging**